lib/screenplay/outfront/sftp.ex

```diff
@@ -2,102 +2,58 @@ defmodule Screenplay.Outfront.SFTP do
   @moduledoc """
   This module handles the CRUD functions to the SFTP server
   """
+  @dialyzer {:no_match, start_connection: 1, write_image: 5, delete_image: 4}
```

Comment:
What were the errors here? I don't have this repo set up on my machine yet >.>

Reply:
`Pattern will never match` errors for the error cases in these functions. The `FakeSFTPClient` doesn't really any errors so when dialyzer complains while that is the module configured for local development.

Reply:
Hmm... I wonder if there's a way to fix that while preserving the warnings for the case where we actually do forget to handle a match. If we explicitly spec the function in the SFTP client as returning `:ok | {:error, ...}`, does that then result in an overspec error?

Reply:
Unfortunately, dialyzer seems to ignore the spec and looks at what the function actually returns. So I would either need to fake an error return for these or ignore the warnings.

Reply:
> Unfortunately, dialyzer seems to ignore the spec and looks at what the function actually returns

Well, normally I'd say that's _fortunate_ behavior, but 😅

I wonder why this hasn't been a problem in the past when I've had fake versions of services that e.g. always returned `:ok`. I think the setup I've used was where the real and fake versions shared a `@behaviour`? But anyway, I don't think it's super important to get to the bottom of this right now unless we've got extra time.

Reply:
Funnily enough, I did try using a `behaviour` and it still yelled about it. I wonder if it's because the "real" module we are using is not ours. I'm not entirely sure, but I'm not super worried about ignoring the warnings specifically for these functions.